overlord/snapstate/backend.go

```diff
@@ -46,15 +48,17 @@ type managerBackend interface {
 	UnlinkSnap(instSnapPath string, meter progress.Meter) error
 	RemoveSnapSecurity(instSnapPath string) error
 	RemoveSnapFiles(instSnapPath string, meter progress.Meter) error
-	RemoveSnapData(name, version string) error
+	RemoveSnapData(name string, revision int) error
 
 	// TODO: need to be split into fine grained tasks
 	Update(name, channel string, flags int, meter progress.Meter) error
-	Rollback(name, ver string, meter progress.Meter) (string, error)
 	Activate(name string, active bool, meter progress.Meter) error
+	// XXX: this one needs to be revno based as well
+	Rollback(name, ver string, meter progress.Meter) (string, error)
 
 	// info
 	ActiveSnap(name string) *snap.Info
+	SnapByNameAndVersion(name, version string) *snap.Info
```

Comment:
@niemeyer name and version is still what we get from the relevant commands, we need to change  snap and api to fix that, we should but doesn't quite belong to this branch that is big enough as it is